src/runtime_src/doc/toc/xrt_native_apis.rst

```diff
@@ -673,3 +676,328 @@ The Run handle/object supports few other use-cases.
+                   // Do something 
+              }
+              else  // Graph is done, quit the loop
+                  break;
+             }
+
+
+Infinite Graph Execution
+************************
+
+The graph runs infinitely if ``xrtGraphRun`` is called with iteration argument -1. While a graph running infinitely the APIs ``xrtGraphWait``, ``xrtGraphSuspend`` and xrtGraphEnd can be used to suspend/end the graph operation after some number of AIE cycles. The API ``xrtGraphResume`` is used to execute the infinitely running graph again. 
+
+
+.. code:: c
+      :number-lines: 39
+           
+           // start from reset state
+           xrtGraphReset(graphHandle);
+           
+           // run the graph infinitely
+           xrtGraphRun(graphHandle, -1);
+           
+           xrtGraphWait(graphHandle,3);  // Suspends the graph after 3 AIE cycles from the previous start 
+           
+           
+           xrtGraphResume(graphHandle); // Restart the suspended graph again to run forever
+           
+           xrtGraphSuspend(graphHandle); // Suspend the graph immediately
+           
+           xrtGraphResume(graphHandle); // Restart the suspended graph again to run forever
+           
+           xrtGraphEnd(graphHandle,5);  // End the graph operation after 5 AIE cycles from the previous start
+
+
+In the example above
+
+- The API ``xrtGraphRun(graphHandle, -1)`` is used to execute the graph infinitely
+- The API ``xrtGraphWait`` suspends the graph after 3 AIE cycles from the graph starts. 
+
+       - If the graph was already run more than 3 AIE cycles the graph is suspended immediately. 
+- The API ``xrtGraphResume`` is used to restart the suspended graph
+- The API ``xrtGraphSuspend`` is used to suspend the graph immediately
+- The API ``xrtGraphEnd(graphHandle,5)`` is  ending the graph after 5 AIE cycles from the previous graph start. 
+       
+       - If the graph was already run more than 5 AIE cycles the graph is suspended immediately.
+       - Using ``xrtGraphEnd`` eliminates the capability of rerunning the Graph (without loading PDI and a graph reset again). 
+
+
+Measuring AIE cycle consumed by the Graph
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The API ``xrtGraphTimeStamp`` can be used to determine AIE cycle consumed between a graph start and stop. 
+
+Here in this example, the AIE cycle consumed by 3 iteration is calculated
+ 
+
+.. code:: c++
+      :number-lines: 35
+           
+           // start from reset state
+           xrtGraphReset(graphHandle);
+           
+           uint64_t begin_t = xrtGraphTimeStamp(graphHandle);
+           
+           // run the graph for 3 iteration
+           xrtGraphRun(graphHandle, 3);
+           
+           xrtGraphWait(graphHandle, 0); 
+           
+           uint64_t end_t = xrtGraphTimeStamp(graphHandle);
+           
+           std::cout<<"Number of AIE cycles consumed in the 3 iteration is: "<< end_t-begin_t; 
```

Comment:
There would be no `std::cout` here, I would miss the fact this is no plain old C API... :-/